model_analyzer/perf_analyzer/perf_config.py

```diff
@@ -95,6 +95,13 @@ def update_config(self, params=None):
             for key in params:
                 self[key] = params[key]
 
+    @classmethod
+    def from_dict(cls, perf_config_dict):
+        perf_config = PerfAnalyzerConfig()
+        for key, val in perf_config_dict.items():
+            setattr(perf_config, key, val)
```

Comment:
Also, I think it would be good to not use `setattr`. You can extract the fields that you need from the dictionary and construct the PerfAnalyzerConfig using them.

Reply:
Is there a reason you think I should't use setattr? It might be more explicit but it also introduces one more place to make changes if I we need to edit the PerAnalyzerConfig.